zerver/tests/test_message_fetch.py

```diff
@@ -2671,6 +2675,92 @@ def _update_tsvector_index(self) -> None:
             """
             )
 
+    def test_get_visible_messages_with_narrow_with(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        othello = self.example_user("othello")
+        self.login("hamlet")
+
+        # Test narrow with accessible stream message operand.
+        self.subscribe(self.example_user("hamlet"), "Scotland")
+
+        message_ids = [
+            self.send_stream_message(self.example_user("iago"), "Scotland") for i in range(5)
+        ]
+
+        narrow = [
+            dict(operator="channel", operand="Scotland"),
+            dict(operator="topic", operand="test"),
+            dict(operator="with", operand=f"{message_ids[0]}"),
+        ]
+        self.message_visibility_test(narrow, message_ids, 2)
+
+        # Test narrow with inaccessible stream message operand.
+        narrow = [
+            dict(operator="channel", operand="Scotland"),
+            dict(operator="topic", operand="test"),
+            dict(operator="with", operand="999999"),
+        ]
+        self.message_visibility_test(narrow, message_ids, 2)
```

Comment:
So this block here seems to be the main bit of tests for the actual logic of the `with` feature. It doesn't feel like you've thought carefully about what you're actually trying to test; no messages are moved in the setup for this test, and so you're not exercising the core logic that we want to verify at all -- that it follows messages that are moved.

So a good version of this part of the test would include in its test setup moving only some of the messages from one topic, and check which messages IDs are returned if you pass the `with` operator using one of the message IDs that was moved vs. one of those that wasn't.

You'll also probably want to test what happens if invalid channel/topic values are sent along with the `with` operator for a real message ID -- which I believe we've specified as doing the same thing as a solo `with` operator, namely just returning the messages in whatever conversation the `with` message was in.

I think you also need to take some time to read https://zulip.readthedocs.io/en/stable/testing/philosophy.html#integration-testing-or-unit-testing -- the very first thing one should do writing tests for a new feature is make sure you've written a test verifying the new functionality, not doing that as an afterthought to just making CI pass via getting coverage for error conditions.